File: cs50/sentimental-credit/credit.py
from cs50 import get_string
credit_number=get_string("Number: ")
crd_rev=credit_number[::-1]
sum=0
for i in range(len(credit_number)):
    if i%2!=0:
        add=int(crd_rev[i])*2
        if add>=10:
            sum+=add//10+(add%10)
        else:
            sum+=add
    else:
        sum+=int(crd_rev[i])
sum10=sum%10
crd_len=len(credit_number)
if sum10==0 and credit_number[:2] in ["34","37"] and crd_len==15:
    print("AMEX")
elif sum10==0 and credit_number[:1]=="4" and crd_len in [13, 16]:
    print("VISA")
elif sum10==0 and credit_number[:2] in ["51","52","53","54","55"] and crd_len==16:
    print("MASTERCARD")
else:
    print("INVALID")

File: cs50/credit/credit.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    long credit_num = get_long("Number: ");
    char crdt[16];
    sprintf(crdt, "%ld", credit_num);
    int sum = 0;
    int len = strlen(crdt);
    for (int i = len - 1, j = 0; i >= 0; i--, j++)
    {
        int digit = crdt[i] - 48;
        if (j % 2 == 1)
        {
            int sumadder = digit * 2;
            if (sumadder >= 10)
            {
                sum += sumadder % 10 + sumadder / 10;
            }
            else
            {
                sum += sumadder;
            }
        }
        else
        {
            sum += digit;
        }
    }
    int digit = crdt[0] - 48;
    if (digit == 4 && sum % 10 == 0 && (len == 16 || len == 13))
    {
        printf("VISA\n");
        return 0;
    }
    digit *= 10;
    digit += crdt[1] - 48;

    if ((digit == 34 || digit == 37) && sum % 10 == 0 && len == 15)
    {

        printf("AMEX\n");
        return 0;
    }
    else if ((digit == 51 || digit == 52 || digit == 53 || digit == 54 || digit == 55) &&
             sum % 10 == 0 && len == 16)
    {
        printf("MASTERCARD\n");
        return 0;
    }
    else
    {
        printf("INVALID\n");
    }
    return 0;
}
